src/modeler/AutoRest.Swagger/Model/ValidationUtilities.cs

```diff
@@ -262,20 +262,23 @@ public static bool IsArrayTypeResponseOperation(Operation op, ServiceDefinition
             if (op.Responses["200"]?.Schema?.Reference?.Equals(string.Empty) == false)
             {
                 var modelLink = op.Responses["200"].Schema.Reference;
+
```

Comment:
This makes me super uncomfortable. There could be a hundred different references like so and sure we should check for the keys in definitions but I'd argue we should probably be throwing an error right when we construct serviceDefinition. Worst case, I'd right a validation rule that runs before any other validation rule and simply checks for sanity of the entire service definition.